client/app/services/data-source.js

```diff
@@ -3,7 +3,7 @@ function DataSource($resource) {
     get: { method: 'GET', cache: false, isArray: false },
     query: { method: 'GET', cache: false, isArray: true },
     test: { method: 'POST', cache: false, isArray: false, url: 'api/data_sources/:id/test' },
-    getSchema: { method: 'GET', cache: true, isArray: true, url: 'api/data_sources/:id/schema' },
+    getSchema: { method: 'GET', cache: false, isArray: true, url: 'api/data_sources/:id/schema' },
```

Comment:
Got it. Let's keep as you did then and revisit if we fill the cache is really necessary.